app/s3_client/__init__.py

```diff
@@ -29,6 +31,17 @@ def get_s3_object(
     )
     s3 = session.resource("s3", config=AWS_CLIENT_CONFIG)
     obj = s3.Object(bucket_name, filename)
+    # This 'proves' that use of moto in the relevant tests in test_send.py
+    # mocks everything related to S3.  What you will see in the logs is:
+    # Exception: CREATED AT <MagicMock name='resource().Bucket().creation_date' id='4665562448'>
+    #
+    # raise Exception(f"CREATED AT {_s3.Bucket(bucket_name).creation_date}")
+    if os.getenv("NOTIFY_ENVIRONMENT") == "test":
+        teststr = str(s3.Bucket(bucket_name).creation_date).lower()
+        if "magicmock" not in teststr:
+            raise Exception(
+                f"xxxxxtest not mocked, use @mock_aws creation date is {teststr}"
```

Comment:
It looks like this exception message is trying to say two different things; is this supposed to just be referencing the current test?  And if so, is there way to perhaps reference the current test this is running in?  `pytest` might provide a way to get that.

Reply:
If you change line 41 to say "in teststr" instead of "not in teststr" and run the tests, you'll see the exception is thrown for each individual test, so it's quite clear which tests are not properly mocked.

I will clean up the line, it doesn't need 'xxxxx'.  Also, I'm showing the creation date, which contains MagicMock if properly mocked.  But I'm not sure that's useful.

Reply:
Thank you for clarifying!